src/model_params.jl

```diff
@@ -0,0 +1,295 @@
+mutable struct ModelParams
```

Comment:
How about renaming `ModelParams` as `SolverInstance`? As this is mostly an instance storing all the parameters we need to solve the problem